### ISSUE TITLE
Heroku app sensitive_config_vars logged in terraform plan under all_config_vars

### ISSUE THREAD

**Issue by drn** [terraform-providers/terraform-provider-heroku#205]
### Terraform Version

```
Terraform v0.12.2
+ provider.aws v2.15.0
+ provider.heroku v2.0.0
```

### Affected Resource(s)

- `heroku_app`, specifically `all_config_vars` with relation to `senstive_config_vars`

### Terraform Configuration Files

Somewhat redacted:

```hcl
resource "heroku_app" "thanx-..." {
  region = "us"
  organization {
    name = "thanx-..."
  }
  name = "thanx-..."
  acm = true
  config_vars = {
    RACK_ENV                  = "..."
    RAILS_ENV                 = "..."
    ...
  }
  sensitive_config_vars = {
    AWS_REGION            = "..."
    AWS_ACCESS_KEY_ID     = "..."
    AWS_SECRET_ACCESS_KEY = "..."
    DATABASE_URL          = "..."
    ...
  }
  stack = "heroku-18"
  buildpacks = ["heroku/nodejs", "heroku/ruby"]
}
```

### Debug Output

Also, somewhat redacted:

```hcl
  # heroku_app.thanx-... will be updated in-place
  ~ resource "heroku_app" "thanx-..." {
        acm                   = true
        all_config_vars       = {
            "AWS_REGION"                  = "..."
            "AWS_ACCESS_KEY_ID"           = "..."
            "AWS_SECRET_ACCESS_KEY"       = "..."
            "DATABASE_URL"                = "..."
            "RACK_ENV"                    = "..."
            "RAILS_ENV"                   = "..."
            ...
        }
        buildpacks            = [
            "heroku/nodejs",
            "heroku/ruby",
        ]
      ~ config_vars           = {
          + "RACK_ENV"                  = "..."
          + "RAILS_ENV"                 = "..."
          ...
        }
        git_url               = "..."
        heroku_hostname       = "..."
        id                    = "..."
        internal_routing      = false
        name                  = "thanx-..."
        region                = "us"
      ~ sensitive_config_vars = (sensitive value)
        stack                 = "heroku-18"
        uuid                  = "..."
        web_url               = "..."

        organization {
            locked   = false
            name     = "thanx-..."
            personal = false
        }
    }
```

### Expected Behavior

Variables included in the `sensitive_config_vars` attribute should not be logged in `terraform plan` under `all_config_vars`.

### Actual Behavior

They were logged under `all_config_vars`.

### Steps to Reproduce

Update `sensitive_config_vars` of an existing `heroku_app` resource (created with Terraform `v0.11.11`)

**Comment by davidji99** [comment]
Hi @drn,

A bit of background:

`heroku_app.config_vars` and `heroku_app.sensitive_config_vars` are essentially both heroku app config vars. So key/value pairs in either attribute both get applied to [/apps/{app_id_or_name}/config-vars](https://devcenter.heroku.com/articles/platform-api-reference#config-vars). As `heroku_app.all_config_vars` is computed, it'll contains all config vars for a given heroku app. 

Furthermore, Heroku does not have a 'sensitivity' distinction for its app config variables. Therefore, the platform API does not return such information. This distinction between `heroku_app.config_vars` and `heroku_app.sensitive_config_vars` is only made during terraform `plan` and `apply` to avoid leaking sensitive data in the console output. The same applies for the same set of config var attributes for the resource `heroku_app_config_association`.

In short, `heroku_app.config_vars` and `heroku_app.sensitive_config_vars` are config vars so `heroku_app.all_config_vars` contains both of those attributes' variables, plus any variables set outside of terraform on an app.

It could be possible to restrict `heroku_app.all_config_vars` to just variables defined in `heroku_app.config_vars`, but that would defeat `heroku_app.all_config_vars`'s purpose of containing all of the app's config variables.

**Comment by drn** [comment]
Hi @davidji99, thanks for the response. Appreciate the details and I completely understand these limitations.

That said, it seems like the distinction of `sensitive_config_vars` from `config_vars` is being rendered pointless by `all_config_vars`, since on `terraform plan`, the `all_config_vars` are displaying variables marked as "sensitive" by `sensitive_config_vars`. It may not be obvious from the initial description since I replaced all the variables with `...` - I'll update the debug output to make it a little more clear what I'm talking about:

`heroku.tf` contents:
```hcl
resource "heroku_app" "thanx-..." {
  region = "us"
  organization {
    name = "thanx-..."
  }
  name = "thanx-..."
  acm = true
  config_vars = {
    RACK_ENV                  = "rack_env"
    RAILS_ENV                 = "rails_env"
    ...
  }
  sensitive_config_vars = {
    AWS_REGION            = "region"
    AWS_ACCESS_KEY_ID     = "access_key_id"
    AWS_SECRET_ACCESS_KEY = "secret_access_key"
    DATABASE_URL          = "database_url"
    ...
  }
  stack = "heroku-18"
  buildpacks = ["heroku/nodejs", "heroku/ruby"]
}
```

`terraform plan` output:
```hcl
  # heroku_app.thanx-... will be updated in-place
  ~ resource "heroku_app" "thanx-..." {
        acm                   = true
        all_config_vars       = {
            "AWS_REGION"                  = "region"
            "AWS_ACCESS_KEY_ID"           = "access_key_id"
            "AWS_SECRET_ACCESS_KEY"       = "secret_access_key"
            "DATABASE_URL"                = "database_url"
            "RACK_ENV"                    = "rack_env"
            "RAILS_ENV"                   = "rails_env"
            ...
        }
        buildpacks            = [
            "heroku/nodejs",
            "heroku/ruby",
        ]
      ~ config_vars           = {
          + "RACK_ENV"                  = "..."
          + "RAILS_ENV"                 = "..."
          ...
        }
        git_url               = "..."
        heroku_hostname       = "..."
        id                    = "..."
        internal_routing      = false
        name                  = "thanx-..."
        region                = "us"
      ~ sensitive_config_vars = (sensitive value)
        stack                 = "heroku-18"
        uuid                  = "..."
        web_url               = "..."

        organization {
            locked   = false
            name     = "thanx-..."
            personal = false
        }
    }
```

Perhaps a solution here is to redact the values in `all_config_vars` if there are corresponding keys in `sensitive_config_vars`. That will not cover all cases where there are sensitive config vars not managed by terraform, but would at least make `sensitive_config_vars` function as expected.

If not, is there a way to suppress `all_config_vars` from the `terraform plan` and `terraform apply` output?

In its current state, `sensitive_config_vars` is not functioning as expected since the sensitive values are being exposed in `terraform plan` under the `all_config_vars` attribute.

Please let me know how I can help out. Happy to help with a solution to resolve

**Comment by mars** [comment]
Totally agree that it's non-intuitive to leak "sensitive" values like this 👍

> Perhaps a solution here is to redact the values in all_config_vars if there are corresponding keys in sensitive_config_vars.

Opened #206 to fix this issue.

**Comment by davidji99** [comment]
HI @drn,

Totally get what you're saying. While `sensitive_config_vars` makes variables hidden, `all_config_vars` exposes those variables. 

Looking at @mars #206, his change is the best course of action in this scenario. Marking `all_config_vars` as `Sensitive` would also help prevent out of band config vars from appearing in the `plan` that shouldn't be exposed in plaintext. After all, it is probably best that the provider take steps to reduce plaintexting secrets at the expense of convenience.

Oh just an fyi, the below wouldn't work because sensitive variables defined in the `heroku_app_config_association` would show up in `heroku_app.all_config_vars`. It wouldn't be possible to filter across resources as far as I'm aware:
> Perhaps a solution here is to redact the values in `all_config_vars` if there are corresponding keys in `sensitive_config_vars`. That will not cover all cases where there are sensitive config vars not managed by terraform, but would at least make `sensitive_config_vars` function as expected.

**Comment by drn** [comment]
Thanks for the quick action, @mars & @davidji99!

Regarding `heroku_app_config_association` - I wasn't actually aware of this resource type. That makes sense.

Cheers

**Comment by mars** [comment]
🏁 [v2.0.1](https://github.com/terraform-providers/terraform-provider-heroku/releases/tag/v2.0.1) has been released, fixing this issue.
